src/stream/src/common/table/state_table.rs

```diff
@@ -796,16 +797,20 @@ fn get_second<T, U>(arg: StreamExecutorResult<(T, U)>) -> StreamExecutorResult<U
 // Iterator functions
 impl<S: StateStore, W: WatermarkBufferStrategy> StateTable<S, W> {
     /// This function scans rows from the relational table.
-    pub async fn iter(&self) -> StreamExecutorResult<RowStream<'_, S>> {
-        self.iter_with_pk_prefix(row::empty()).await
+    pub async fn iter(&self, exhaust_iter: bool) -> StreamExecutorResult<RowStream<'_, S>> {
```

Comment:
> Changing `iter` related interfaces in state table and storage table requires changing executors. To avoid further changing the interfaces in the future, I suggest we pass a struct (`PrefetchOptions` or `IterOptions` with `limit_hint`, `enable_prefetch`, ... inside) instead in state table and storage table interfaces.

fixed